apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx

```diff
@@ -118,23 +132,97 @@ const SurveyResultsHeader = React.createClass({
-                <td>{allWorkshopsForCourse[row['key']]}</td>
-              </tr>
-            );
-          })
-        }
-      </tbody>
+      <table className="table table-bordered" style={{width: 'auto'}}>
+        <thead>
+          <tr>
+            <th/>
+            <th>
+              {this.state.workshopName}
+            </th>
+            <th>
+              Across all my workshops for {this.state.selectedCourse}
+            </th>
+            <th>
+              Across all workshops for {this.state.selectedCourse}
+            </th>
+          </tr>
+        </thead>
+        <tbody>
+          {
+            rowOrder.map(function (row, i) {
+              return (
+                <tr key={i}>
+                  <td style={row['heading'] ? styles.questionGroupHeader : styles.individualQuestion}>{row['text']}</td>
+                  <td>{thisWorkshop[row['key']]}</td>
+                  <td>{allMyWorkshopsForCourse[row['key']]}</td>
+                  <td>{allWorkshopsForCourse[row['key']]}</td>
+                </tr>
+              );
+            })
+          }
+        </tbody>
+      </table>
+    );
+  },
+
+  renderOrganizerSurveyResults() {
+    let allWorkshopsForCourse = this.state.workshopSurveyData['all_workshops_for_course'];
+    let allMyWorkshopsForCourse = this.state.workshopSurveyData['all_my_workshops_for_course'];
+    let breakoutData = _.omit(this.state.workshopSurveyData, ['all_workshops_for_course', 'all_my_workshops_for_course']);
+
+    let headings = ['', 'Across all workshops for ' + this.state.selectedCourse, 'Across all workshops I organized for ' + this.state.selectedCourse];
+
+    if (this.state.selectedWorkshopId) {
+      headings.push(this.state.workshopName);
+    } else {
+      headings = headings.concat(Object.keys(breakoutData));
+    }
+
+    return (
+      <table className="table table-bordered" style={{width: 'auto'}}>
+        <thead>
+          <tr>
+            {
+              headings.map(function (heading, i) {
+                return (
+                  <th key={i}>
+                    {heading}
+                  </th>
+                );
+              })
+            }
+          </tr>
+        </thead>
+        <tbody>
+          {
+            rowOrder.map(function (row, i) {
+              return (
+                <tr key={i}>
+                  <td style={row['heading'] ? styles.questionGroupHeader : styles.individualQuestion}>{row['text']}</td>
+                  <td>{allWorkshopsForCourse[row['key']]}</td>
+                  <td>{allMyWorkshopsForCourse[row['key']]}</td>
+                  {
+                    headings.slice(3).map(function (heading, i) {
+                      return (
+                        <td key={i}>
+                          {
+                            // TODO: Find a better way to do the below statement
+                            breakoutData[heading] ? breakoutData[heading][row['key']] : breakoutData['this_workshop'][row['key']]
```

Comment:
As discussed offline, if you bind the function to `this` or better use ES6 arrow functions, then you can access and branch on `this.state. selectedWorkshopId`